### ISSUE TITLE
19.04 - Public view dosen't load settings after deleting and load from existing view again

### ISSUE THREAD

**Issue by itsul** [centreon/centreon#8858]
After loading a public view and deleting and loading it again, the settings of the widget are not set for the user who loaded the view. You can repeat the step deleting and loading multiple times with no change of result.

Bug testet on Centreon Versions: 19.04.16 and 19.04.17

For the RPM based systems

-- Copy/Paste the result of the following command --
```
$ rpm -qa | grep centreon
centreon-database-19.04.17-5.el7.centos.noarch
centreon-broker-storage-19.04.0-2.el7.centos.x86_64
centreon-widget-graph-monitoring-19.04.1-1.el7.centos.noarch
centreon-license-manager-common-19.04.1-3.el7.centos.noarch
centreon-engine-19.04.3-1.el7.centos.x86_64
centreon-plugin-Applications-Monitoring-Centreon-Database-20200602-094050.el7.centos.noarch
centreon-trap-19.04.17-5.el7.centos.noarch
centreon-connector-perl-19.04.1-1.el7.centos.x86_64
centreon-widget-live-top10-memory-usage-19.04.1-1.el7.centos.noarch
centreon-plugin-Operatingsystems-Linux-Snmp-20200602-094050.el7.centos.noarch
centreon-plugin-Hardware-Printers-Generic-Snmp-20200602-094050.el7.centos.noarch
centreon-widget-host-monitoring-19.04.5-6.el7.centos.noarch
centreon-common-19.04.17-5.el7.centos.noarch
centreon-poller-centreon-engine-19.04.17-5.el7.centos.noarch
centreon-connector-ssh-19.04.1-1.el7.centos.x86_64
centreon-release-19.04-1.el7.centos.noarch
centreon-engine-daemon-19.04.3-1.el7.centos.x86_64
centreon-widget-tactical-overview-19.04.1-1.el7.centos.noarch
centreon-widget-engine-status-19.04.3-1.el7.centos.noarch
centreon-plugins-base-1.18-2.el7.centos.noarch
centreon-plugin-Applications-Databases-Mysql-20200602-094050.el7.centos.noarch
centreon-plugin-Applications-Protocol-Ldap-20200602-094050.el7.centos.noarch
centreon-plugin-Network-Cisco-Standard-Snmp-20200602-094050.el7.centos.noarch
centreon-plugin-Operatingsystems-Windows-Snmp-20200602-094050.el7.centos.noarch
centreon-plugin-Network-Switchs-Cisco-Smallbusiness-Standard-Snmp-20200602-094050.el7.centos.noarch
centreon-perl-libs-19.04.17-5.el7.centos.noarch
centreon-auto-discovery-server-19.04.4-1.el7.centos.x86_64
centreon-widget-servicegroup-monitoring-19.04.1-1.el7.centos.noarch
centreon-19.04.17-5.el7.centos.noarch
centreon-broker-cbd-19.04.0-2.el7.centos.x86_64
centreon-broker-cbmod-19.04.0-2.el7.centos.x86_64
centreon-license-manager-19.04.1-3.el7.centos.noarch
centreon-plugin-Hardware-Servers-Lenovo-Xcc-Snmp-20200602-094050.el7.centos.noarch
centreon-plugin-Applications-Monitoring-Centreon-Central-20200602-094050.el7.centos.noarch
centreon-plugin-Applications-Monitoring-Centreon-Poller-20200602-094050.el7.centos.noarch
centreon-plugin-Applications-Protocol-Http-20200602-094050.el7.centos.noarch
centreon-web-19.04.17-5.el7.centos.noarch
centreon-pp-manager-19.04.2-2.el7.centos.noarch
centreon-widget-httploader-19.04.1-2.el7.centos.noarch
centreon-widget-hostgroup-monitoring-19.04.1-1.el7.centos.noarch
centreon-clib-19.04.1-1.el7.centos.x86_64
centreon-connector-19.04.1-1.el7.centos.x86_64
centreon-widget-live-top10-cpu-usage-19.04.2-1.el7.centos.noarch
centreon-plugin-Applications-Monitoring-Centreon-Map4-Jmx-20200602-094050.el7.centos.noarch
centreon-plugin-Applications-Protocol-Dns-20200602-094050.el7.centos.noarch
centreon-widget-service-monitoring-19.04.5-6.el7.centos.noarch
centreon-base-config-centreon-engine-19.04.17-5.el7.centos.noarch
centreon-broker-core-19.04.0-2.el7.centos.x86_64
centreon-widget-grid-map-19.04.1-1.el7.centos.noarch
centreon-engine-extcommands-19.04.3-1.el7.centos.x86_64
centreon-plugin-Applications-Protocol-Ftp-20200602-094050.el7.centos.noarch
centreon-broker-19.04.0-2.el7.centos.x86_64
centreon-plugin-Hardware-Ups-Standard-Rfc1628-Snmp-20200602-094050.el7.centos.noarch
centreon-plugin-Hardware-Storage-Hp-Msa2000-Snmp-20200602-094050.el7.centos.noarch

***Operating System***
CentOS

***Browser used***

- [X ] Chromium 83.0.4103.116
- [X]  Firefox 78.0.1
- [ ] Internet Explorer IE11
- [ ] Safari


### Steps to Reproduce

Please describe precisely the steps to reproduce the encountered issue.

1) View created by user A and marked as "Public"
2) Add standard widget like Service Monitoring
3) Adjust settings in the widget
4) With other user B add the view (load from existing view) -> settings of widgets are saved, everything works
5) Delete the view with user B
6) Add the view again with user B -> settings of widgets aren't loaded, the widget are useless

You can do step 5 & 6 multiple times, dosen't change the end result. (Testet on two independent 19.04 machines)

If delete the view manually with a MySQL Command like this "delete from custom_view_user_relation where user_id=B_USERID;" everything is normal again until you load, delete and load it again. It seems the delete button only changes the flag is_consumed to 1.

### Describe the received result
Settings of the widget for user B are not set, widget for user A is fine

### Describe the expected result
Settings of the widget should be the same like user A

### Logs
Nothing

**Comment by cgagnaire** [comment]
Hi @itsul,
A PR has been open for this bug, but the fix should not reach 19.04.x maintenance branch.
I'll let @Moujimouja settle this.

**Comment by kduret** [comment]
Note : If you apply the fix on your platform, you'll need to : 

- Login centreon web with the user who share the custom view
- Switch to the custom view with broken preferences for other users
- Click on "Share view", and then click on "Share"

==> This will restore preferences for other users
